server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/NettyServerHandler.scala

```diff
@@ -20,11 +20,44 @@ import sttp.tapir.server.netty.NettyResponseContent.{
 import sttp.tapir.server.netty.{NettyResponse, NettyServerRequest, Route}
 
 import scala.collection.JavaConverters._
+import scala.collection.mutable.{Queue => MutableQueue}
+import scala.concurrent.Future
+import scala.util.Failure
+import scala.util.Success
+import scala.util.control.NonFatal
+import scala.concurrent.ExecutionContext
 
-class NettyServerHandler[F[_]](route: Route[F], unsafeRunAsync: (() => F[Unit]) => Unit, maxContentLength: Option[Int])(implicit
+/** @param unsafeRunAsync
+  *   Function which dispatches given effect to run asynchronously, returning its result as a Future, and function of type `() =>
+  *   Future[Unit]` allowing cancellation of that Future. For example, this can be realized by
+  *   `cats.effect.std.Dispatcher.unsafeToFutureCancelable`.
+  */
+class NettyServerHandler[F[_]](
+    route: Route[F],
+    unsafeRunAsync: (() => F[ServerResponse[NettyResponse]]) => (Future[ServerResponse[NettyResponse]], () => Future[Unit]),
+    maxContentLength: Option[Int]
+)(implicit
     me: MonadError[F]
 ) extends SimpleChannelInboundHandler[HttpRequest] {
 
+  // By using the Netty event loop assigned to this channel we get two benefits:
```

Comment:
let's maybe mention here that this is copied from http4's code, just to maintain proper attribution :)